src/Microsoft.DotNet.Build.Tasks/PackageFiles/OptionalTooling.targets

```diff
@@ -0,0 +1,135 @@
+                                 PackagesDir="$(PackagesDir)" />
+  </Target>
+
+  <!--
+    Restore optional tooling by generating the XML string of a NuGet.Config with plaintext
+    credentials, then injecting it into "dotnet restore" using /dev/stdin.
+    Not supported on Windows.
+  -->
+  <Target Name="RestoreOptionalToolingUnencrypted"
+          DependsOnTargets="CreateOptionalRestoreFeedItems;
+                            PrepareOptionalToolProjectJson"
+          Condition="'$(MSBuildRuntimeType)'=='core'">
+    <PropertyGroup>
+      <!-- Create a new restore command with no source parameters, only configfile. -->
+      <OptionalRestoreCommand>"$(DnuToolPath)"</OptionalRestoreCommand>
+      <OptionalRestoreCommand>$(OptionalRestoreCommand) restore</OptionalRestoreCommand>
+      <OptionalRestoreCommand Condition="'$(UseNuGetHttpCache)'!='true'">$(OptionalRestoreCommand) --no-cache</OptionalRestoreCommand>
+      <OptionalRestoreCommand>$(OptionalRestoreCommand) --packages "$(PackagesDir.TrimEnd('/\'.ToCharArray()))"</OptionalRestoreCommand>
+      <OptionalRestoreCommand>$(OptionalRestoreCommand) --configfile $(OptionalRestoreConfigPath)</OptionalRestoreCommand>
+    </PropertyGroup>
+
+    <Error Text="Restoring optional tooling with in-memory credentials is not supported on Windows. Use the desktop framework for the encrypted credential flow."
+           Condition="'$(OS)'=='Windows_NT'" />
+
+    <Error Text="Restoring optional tooling with in-memory credentials is not yet implemented. No cross-platform optional tooling exists yet." />
+  </Target>
+
+  <!--
+    Cleanup steps: try to ensure non-authenticated restore attempts don't attempt to restore
+    optional packages and reduce credential exposure to the remaining steps in the build.
+  -->
+  <Target Name="CleanOptionalToolingRestore"
+          Condition="'$(SkipCleanOptionalToolingRestore)'!='true'">
+    <ItemGroup>
+      <OptionalRestoreTemporaryFilePath Include="$(OptionalRestoreConfigPath);
+                                                 $(OptionalToolingProjectJsonPath)" />
+    </ItemGroup>
+
+    <Delete Files="@(OptionalRestoreTemporaryFilePath)" Condition="Exists('%(Identity)')" />
+  </Target>
+
+  <!--
+    Resolve tooling from the optional restore lockfile. Other targets can find optional tooling
+    in ResolvedOptionalToolReferences.
+  -->
+  <Target Name="ResolveOptionalTools"
+          DependsOnTargets="GetOptionalToolingPaths">
+
+    <Error Text="Optional tooling has not been restored. Use the 'RestoreOptionalToolingPackages' target first."
+           Condition="!Exists('$(OptionalToolingProjectLockJsonPath)')" />
+
+    <ItemGroup>
+      <OptionalToolingTargetMoniker Include=".NETFramework,Version=v4.5" />
+    </ItemGroup>
+
+    <PrereleaseResolveNuGetPackageAssets ProjectLockFile="$(OptionalToolingProjectLockJsonPath)"
+                                         NuGetPackagesDirectory="$(PackagesDir)"
```

Comment:
I think this is a good fix. Pretty sure it works in master because newer versions of nuget put the package dir in the lockfile, which means https://github.com/dotnet/buildtools/blob/4f918b5f338f0cac04b29cd4bfe17acfd5d8e1b9/src/Microsoft.DotNet.Build.Tasks/PrereleaseResolveNuGetPackageAssets.cs#L267 can read it. If you see that entry in the 1.1.0 optional tooling lockfile you could try to port that, but the fix you have here seems like a much safer bet.